src/lib.rs

```diff
@@ -0,0 +1,1210 @@
+            return None;
+        }
+        // The subtraction can't underflow because self.len > 0.
+        let v = self.get(self.len - 1);
+        debug_assert!(v.is_some());
+        self.len -= 1;
+        self.mask_last_block();
+        v
+    }
+
+    /// Returns the number of elements in the Vob.
+    pub fn len(&self) -> usize {
+        self.len
+    }
+
+    /// Returns true if the Vob has a length of 0.
+    ///
+    /// # Examples
+    /// ```
+    /// use vob::Vob;
+    /// assert_eq!(Vob::from_elem(2, true).is_empty(), false);
+    /// assert_eq!(Vob::new().is_empty(), true);
+    /// ```
+    pub fn is_empty(&self) -> bool {
+        self.len == 0
+    }
+
+    /// Splits the collection into two at the given index.
+    ///
+    /// Returns a newly allocated Self. self contains elements [0, at), and the returned Self
+    /// contains elements [at, len).
+    ///
+    /// Note that the capacity of self does not change.
+    ///
+    /// # Examples
+    /// ```
+    /// use vob::Vob;
+    /// let mut v1 = Vob::new();
+    /// v1.push(true);
+    /// v1.push(false);
+    /// let v2 = v1.split_off(1);
+    /// assert_eq!(v1, Vob::from_elem(1, true));
+    /// assert_eq!(v2, Vob::from_elem(1, false));
+    /// ```
+    pub fn split_off(&mut self, at: usize) -> Vob<T> {
+        if at >= self.len {
+            return Vob::<T>::new_with_storage_type(0);
+        }
+        let mut nv = Vob::<T>::new_with_storage_type(self.len - at);
+        // This could easily be made more efficient.
+        for b in self.iter().skip(at) {
+            nv.push(b);
+        }
+        self.len = at;
+        self.mask_last_block();
+        nv
+    }
+
+    /// Returns the value of the element at position `index` or `None` if out of bounds.
+    ///
+    /// # Examples
+    /// ```
+    /// use vob::Vob;
+    /// let mut v = Vob::new();
+    /// v.push(false);
+    /// assert_eq!(v.get(0), Some(false));
+    /// assert_eq!(v.get(1), None);
+    /// ```
+    pub fn get(&self, index: usize) -> Option<bool> {
+        if index >= self.len {
+            return None;
+        }
+        let b = self.vec[block_offset::<T>(index)];
+        Some(b & (T::one() << (index % bits_per_block::<T>())) != T::zero())
+    }
+
+    /// Sets the value of the element at position `index` or `None` if out of bounds. Returns
+    /// `true` if this led to a change in the underlying storage or `None` if out of bounds.
+    ///
+    /// # Examples
+    /// ```
+    /// use vob::Vob;
+    /// let mut v = Vob::new();
+    /// v.push(false);
+    /// v.set(0, true);
+    /// assert_eq!(v.get(0), Some(true));
+    /// assert_eq!(v.set(0, false), Some(true));
+    /// assert_eq!(v.set(0, false), Some(false));
+    /// ```
+    pub fn set(&mut self, index: usize, value: bool) -> Option<bool> {
+        if index >= self.len {
+            return None;
+        }
+        let m = T::one() << (index % bits_per_block::<T>());
+        let b = block_offset::<T>(index);
```

Comment:
So the **o**ffset is called `b` and the **b**lock is called `o`?

Are you sure we want these short variable names? Without decent names, I can only store about 2 or 3 in my head. They are really hard to read.

Reply:
This is a bad example, I admit.

Reply:
[i.e. I used the wrong names.]

Reply:
Okie dokie.

[Have you been hanging around with physicists again?] :microscope: